lib/cfg.c

```diff
@@ -329,6 +329,24 @@ cfg_new(gint version)
   return self;
 }
 
+GlobalConfig *
+cfg_load_config_snippet(const gchar *fname)
+{
+  GlobalConfig *cfg = cfg_new(0);
```

Comment:
If its in a relation to a real config,  its version should be propagated here.

But, due to the hackish nature of this patch i was wondering if it works at all. Since globalconfig is such a monster, its not always clear what will happen in a case like this.